src/fiatExchanges/PaymentMethodSection.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, LayoutAnimation, StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import Expandable from 'src/components/Expandable'
+import Touchable from 'src/components/Touchable'
+import {
+  CicoQuote,
+  getFeeValueFromQuotes,
+  PaymentMethod,
+  ProviderQuote,
+  SimplexQuote,
+} from 'src/fiatExchanges/utils'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+
+interface PaymentMethodSectionProps {
+  paymentMethod: PaymentMethod
+  cicoQuotes: CicoQuote[]
+  setNoPaymentMethods: React.Dispatch<React.SetStateAction<boolean>>
+  quoteOnPress: (cicoQuote: CicoQuote) => void
+}
+
+export function PaymentMethodSection({
+  paymentMethod,
+  cicoQuotes,
+  setNoPaymentMethods,
+  quoteOnPress,
+}: PaymentMethodSectionProps) {
+  const { t } = useTranslation()
+  const sectionQuotes = cicoQuotes.filter(({ quote }) => quote.paymentMethod === paymentMethod)
+  const localCurrency = useSelector(getLocalCurrencyCode)
+
+  const isExpandable = sectionQuotes.length > 1
+  const [expanded, setExpanded] = useState(false)
+
+  const toggleExpanded = () => {
+    LayoutAnimation.easeInEaseOut()
+    setExpanded(!expanded)
+  }
+  if (!sectionQuotes.length) {
+    setNoPaymentMethods(true)
+    return null
+  }
+  const renderFeeAmount = (quote: SimplexQuote | ProviderQuote, postFix: string) => {
+    const feeAmount = getFeeValueFromQuotes(quote)
+
+    if (feeAmount === undefined) {
+      return null
+    }
+
+    return (
+      <Text>
+        <CurrencyDisplay
+          amount={{
+            value: 0,
+            localAmount: {
+              value: feeAmount,
+              currencyCode: localCurrency,
+              exchangeRate: 1,
+            },
+            currencyCode: localCurrency,
+          }}
+          showLocalAmount={true}
+          hideSign={true}
+          style={styles.fee}
+        />{' '}
+        {postFix}
+      </Text>
+    )
+  }
+  return (
+    <View style={styles.container}>
+      <Touchable
+        onPress={
+          isExpandable
+            ? toggleExpanded
+            : ((quoteOnPress(sectionQuotes[0]) as unknown) as () => void)
+        }
+      >
+        <View>
+          <Expandable
+            arrowColor={colors.greenUI}
+            containerStyle={{
+              ...styles.expandableContainer,
+              paddingVertical: isExpandable ? (expanded ? 22 : 27) : 16,
+            }}
+            isExpandable={isExpandable}
+            isExpanded={expanded}
+          >
+            {isExpandable ? (
```

Comment:
these nested inline ternarys are a bit tricky to follow imo

Reply:
i'll clean it up a bit and separate them out into helper functions